File: src/main/java/com/github/bjoernpetersen/jmusicbot/user/InvalidTokenException.java
package com.github.bjoernpetersen.jmusicbot.user;

public final class InvalidTokenException extends Exception {

  public InvalidTokenException() {
  }

  public InvalidTokenException(String message) {
    super(message);
  }

  public InvalidTokenException(String message, Throwable cause) {
    super(message, cause);
  }

  public InvalidTokenException(Throwable cause) {
    super(cause);
  }
}

File: src/main/kotlin/net/bjoernpetersen/musicbot/api/auth/InvalidTokenException.kt
package net.bjoernpetersen.musicbot.api.auth

/**
 * This exception is thrown if a token has an invalid structure or its signature can't be verified.
 */
open class InvalidTokenException : Exception {
    constructor() : super()
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
    constructor(cause: Throwable) : super(cause)
}

/**
 * Thrown if the token is invalid because its signature didn't match.
 */
class InvalidSignatureException : InvalidTokenException {
    constructor() : super()
    constructor(cause: Throwable) : super(cause)
}
